1-Draft/RFCnnnn-ForEach-Parallel-Implementation.md

```diff
@@ -0,0 +1,110 @@
+---
+RFC: RFCnnnn
+Author: Paul Higinbotham
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Engine
+Comments Due: July 1, 2019
+Plan to implement: Yes
+---
+
+# Implement PowerShell language foreach -parallel
+
+Windows PowerShell currently supports the foreach language keyword with the -parallel switch flag, but only for workflow scripts.
+
+```powershell
+
+workflow wf1 {
+    $list = 1..5
+    foreach -parallel -throttlelimit 5 ($item in $list) {
+        Start-Sleep -Seconds 1
+        Write-Output "Output $item"
+    }
+}
+
+```
+
+This will run the script block with each value in the `$list` array, in parallel using workflow jobs.
+However, workflow is not supported in PowerShell Core 6, partly because it is a Windows only solution but also because it is cumbersome to use.
+In addition the workflow implementation is very heavy weight, using lots of system resources.  
+
+This is a proposal to re-implement `foreach -parallel` in PowerShell Core, using PowerShell's support for concurrency via Runspaces.
+It is similar to the [ThreadJob module](https://www.powershellgallery.com/packages/ThreadJob/1.1.2) except that it becomes part of the PowerShell language via `foreach -parallel`.
+

```

Comment:
Regardless of how implemented (as a cmdlet or a foreach extension), I seriously hope you will make this compatible with psexec. When I filed an issue against threadjob noting that it broke psexec, psexec was blamed. I doubt highly that sysinternals/Mark Russinovich would agree with that.